### PR TITLE
Test to track RAPIDS-side issues re SPARK-32639

### LINKED ISSUE #1463: [FEA] Add Parquet GroupType mapkey field test
[SPARK-32639](https://issues.apache.org/jira/browse/SPARK-32639) fixed a Parquet parsing issue when handling a particular form of map encoding in Parquet.  A sample file is attached to that JIRA.  It would be good to add a plugin unit test that verifies the plugin can load this file properly as well.

### PR DESCRIPTION
**Pull request by gerashegalov**
This PR closes #1463. The test documents the scenario that fails on Spark (CPU) prior to Spark 3.1.0. 

It results in an executor JVM crash on GPU #1576. Since xfail does not handle the process crash gracefully and fails pytest, skip is used instead.

### CODE REVIEW

**Comment by gerashegalov** [comment]
build

File: integration_tests/src/main/python/parquet_test.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
```
Comment (jlowe) [inline integration_tests/src/main/python/parquet_test.py:14]: Need to add 2021 copyrights

**Review by jlowe — CHANGES_REQUESTED** [review]

**Review by jlowe — APPROVED** [review]

**Comment by jlowe** [comment]
build

**Comment by pxLi** [comment]
build

**Pull request merged**